dot/network/inbound.go

```diff
@@ -0,0 +1,73 @@
+package network
+
+import (
+	libp2pnetwork "github.com/libp2p/go-libp2p-core/network"
+)
+
+func (s *Service) readStream(stream libp2pnetwork.Stream, decoder messageDecoder, handler messageHandler) {
+	// we NEED to reset the stream if we ever return from this function, as if we return,
+	// the stream will never again be read by us, so we need to tell the remote side we're
+	// done with this stream, and they should also forget about it.
+	defer s.resetInboundStream(stream)
+	s.streamManager.logNewStream(stream)
+
+	peer := stream.Conn().RemotePeer()
+	msgBytes := s.bufPool.get()
+	defer s.bufPool.put(msgBytes)
+
+	for {
+		tot, err := readStream(stream, msgBytes[:])
```

Comment:
nit maybe call it `n` (aka n bytes from std lib) or `total` / `totalBytes`? `tot` feels strange to me haha!